components/brave_rewards/resources/extension/brave_rewards/content_scripts/reddit.ts

```diff
@@ -0,0 +1,417 @@
+  braveTipIconContainer.style.display = 'inline-block'
+  braveTipIconContainer.style.marginBottom = '-2px'
+  braveTipIconContainer.style.position = 'relative'
+  braveTipIconContainer.style.verticalAlign = 'middle'
+  return braveTipIconContainer
+}
+
+const createRedditTipAction = (isPost: boolean) => {
+  const braveTipAction = document.createElement('div')
+  braveTipAction.className = 'action-brave-tip'
+  if (isPost) {
+    braveTipAction.style.display = 'flex'
+    braveTipAction.style.height = '25px'
+    braveTipAction.style.outline = 'none'
+    braveTipAction.style.borderRadius = '2px'
+  }
+  braveTipAction.setAttribute('data-original-title', getMessage('redditTipsHoverText'))
+  return braveTipAction
+}
+
+const createOldRedditTipAction = () => {
+  const braveTipAction = document.createElement('li')
+  braveTipAction.className = 'action-brave-tip'
+  return braveTipAction
+}
+
+const createRedditTipIcon = () => {
+  const braveTipIcon = document.createElement('span')
+  braveTipIcon.className = 'tip-icon--medium'
+  braveTipIcon.style.background = 'transparent'
+  braveTipIcon.style.content = 'url(\'data:image/svg+xml;utf8,<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 105 100" style="enable-background:new 0 0 105 100;" xml:space="preserve"><style type="text/css">.st1{fill:%23662D91;}.st2{fill:%239E1F63;}.st3{fill:%23FF5000;}.st4{fill:%23FFFFFF;stroke:%23FF5000;stroke-width:0.83;stroke-miterlimit:10;}</style><title>BAT_icon</title><g id="Layer_2_1_"><g id="Layer_1-2"><polygon class="st1" points="94.8,82.6 47.4,55.4 0,82.9 "/><polygon class="st2" points="47.4,0 47.1,55.4 94.8,82.6 "/><polygon class="st3" points="0,82.9 47.2,55.9 47.4,0 "/><polygon class="st4" points="47.1,33.7 28,66.5 66.7,66.5 "/></g></g></svg>\')'
+  braveTipIcon.style.display = 'inline-block'
+  braveTipIcon.style.fontSize = '18px'
+  braveTipIcon.style.fontStyle = 'normal'
+  braveTipIcon.style.height = '16px'
+  braveTipIcon.style.position = 'relative'
+  braveTipIcon.style.verticalAlign = 'baseline'
+  braveTipIcon.style.width = '16px'
+  return braveTipIcon
+}
+
+const createRedditTipActionCount = () => {
+  const braveTipActionCount = document.createElement('span')
+  braveTipActionCount.className = 'reddit-actionCount'
+  braveTipActionCount.style.color = 'inherit'
+  braveTipActionCount.style.display = 'inline-block'
+  braveTipActionCount.style.fontSize = '12px'
+  braveTipActionCount.style.fontWeight = 'bold'
+  braveTipActionCount.style.lineHeight = '1'
+  braveTipActionCount.style.marginLeft = '3px'
+  braveTipActionCount.style.position = 'relative'
+  braveTipActionCount.style.verticalAlign = 'text-bottom'
+  return braveTipActionCount
+}
+
+const createRedditTipActionCountPresentation = () => {
+  const braveTipActionCountPresentation = document.createElement('span')
+  braveTipActionCountPresentation.className = 'ProfileReddit-actionCountForPresentation'
```

Comment:
Just a little bit of backstory here...

When I implemented Twitter tips, I had to do all of our DOM manipulations in the shadow DOM. This meant that I was no longer able to reference classes that were outside of the shadow DOM, hence why I had to include the styles, etc. inline. But I left the original Twitter class names in the code as a sort of "comment" so I could remember which classes the various buttons/labels/etc. were based on.

So, not sure how you want to handle that here, just wanted to explain my original thinking. Just mentioning it because I'm guessing there probably isn't actually a `ProfileReddit-actionCountForPresentation` class in Reddit, so this may be a little bit confusing (even though in the grand scheme of things, the class names have no effect anyway because of the shadow DOM).

Reply:
Yes, I figured these were class names specific to Twitter. I kept the naming simply for a type of convention for the same reasons (to know which elements we were working with quickly)

Reply:
OK, gotcha.